spec/configuration_spec.lua

```diff
@@ -125,6 +125,53 @@ describe('Configuration object', function()
       assert.same(services, filter_services(services, { '42' }))
       assert.same({}, filter_services(services, { '21' }))
     end)
+
+    describe("with service filter", function()
+      local mockservices = {
+        {id="42", hosts={"test.foo.com", "test.bar.com"}},
+        {id="12", hosts={"staging.foo.com"}},
+        {id="21", hosts={"prod.foo.com"}},
+      }
+
+      it("validates default case", function()
+        assert.same(filter_services(mockservices, nil), mockservices)
+      end)
+
+      it("reads from environment variable", function()
```

Comment:
We should probably also test `APICAST_SERVICES_FILTER_BY_URL = ''`. Empty envs can be problematic in some cases.